packages/expo-updates/android/src/main/java/expo/modules/updates/loader/Crypto.java

```diff
@@ -0,0 +1,95 @@
+package expo.modules.updates.loader;
+
+import android.annotation.SuppressLint;
+import android.security.keystore.KeyProperties;
+import android.util.Base64;
+
+import java.io.IOException;
+import java.security.InvalidKeyException;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.Signature;
+import java.security.SignatureException;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.X509EncodedKeySpec;
+
+import okhttp3.CacheControl;
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class Crypto {
+
+  public interface RSASignatureListener {
+    void onError(Exception exception, boolean isNetworkError);
+    void onCompleted(boolean isValid);
+  }
+
+  private static String PUBLIC_KEY_URL = "https://exp.host/--/manifest-public-key";
```

Comment:
Later we'll want to split this class into two so that Expo-proprietary code is less coupled to the standard.